### ISSUE TITLE
0.24.0-beta-2 incorrectly translates strings with `&amp;` for Android as just `&`

### ISSUE THREAD

**Issue by darronschall** [icerockdev/moko-resources#689]
I have a string defined in `commonMain/moko-resources/base/strings.xml` that has this sample string:

```xml
<string name="example__one_and_two">One &amp; Two</string>
```

When the Android code is generated in `build/generated/moko-resources/androidMain/res/values/multiplatform_strings.xml` it gets incorrectly translated as:

```xml
<string name="example__one_and_two">One & Two</string>
```

This yields a compiler error: `The entity name must immediately follow the '&' in the entity reference.` The Android code should retain the `&amp;` since the generated code remains as XML.

This worked correctly in 0.23; it appears to be a 0.24 regression.

NOTE: That iOS generation is correct. It generates as:

```
"example__one_and_two" = "One & Two";
```

**Comment by darronschall** [comment]
I think this might have been introduced by https://github.com/icerockdev/moko-resources/pull/677/files#diff-919d3dea14f8410e534d39477b2ff3e7b1e9c809155c974ca03ee474cfae2fd0L73, when `convertXmlStringToAndroidLocalization` was removed, which indirectly changed from `StringEscapeUtils.escapeXml11(it)` to `StringEscapeUtils.escapeEcmaScript(it)` on Android.

**Comment by tamimattafi** [comment]
Having the same issue

**Comment by tamimattafi** [comment]
I also noticed that the new version seem to be escaping style tags such as `<b>Some text</b>`
The generated code gives someting like this: `<b>*UNICODE*<\/b>` and an error:
```
Error: The content of elements must consist of well-formed character data or markup.
Element b is not closed
Tag name expected
```

**Comment by Alex009** [comment]
fixed in #692 , will be available in 0.24.0-beta-3
